### ISSUE TITLE
cannot unmarshal string into Go struct field TodoTarget.target.id of type int

### ISSUE THREAD

**Issue by josephburnett** [xanzy/go-gitlab#1743]
When mentioned on a commit, the target id will be the commit hash which is a string. The [`TodoTarget.ID`](https://github.com/xanzy/go-gitlab/blob/0886867b3ba7ab3554169fab2b418868c9f016b5/todos.go#L61) is an int so unmarshaling fails.

```json
        "action_name": "mentioned",
        "target_type": "Commit",
        "target": {
            "id": "6a49b920e067e123703578ac38f4819f8e5a2823",
```

```
cannot unmarshal string into Go struct field TodoTarget.target.id of type int
```

**Comment by josephburnett** [comment]
The [GitLab TODO API page](https://docs.gitlab.com/ee/api/todos.html) pretty clearly shows an integer ID. So a string is unexpected.

```json
    "action_name": "marked",
    "target_type": "MergeRequest",
    "target": {
      "id": 34,
```

**Comment by josephburnett** [comment]
Opened an issue with GitLab: https://gitlab.com/gitlab-org/gitlab/-/issues/416180

**Comment by josephburnett** [comment]
Unfortunately we cannot change the type of `ID` because it would be a breaking change. So instead I've implemented a workaround that creates a derived field `IDString`: https://github.com/xanzy/go-gitlab/pull/1744

The GitLab `id` will be routed to either `ID` or `IDString` based on it's JSON type. And either `ID` or `IDString` will be emitted as `id`. So the effect is entirely contained in the library and no visible in the JSON output.

**Comment by svanharmelen** [comment]
Fixed by #1744
